engine/time/src/main/java/io/deephaven/time/calendar/BusinessCalendarXMLParser.java

```diff
@@ -103,12 +122,16 @@ private static BusinessCalendarInputs parseBusinessCalendarInputs(@NotNull final
             Element root = loadXMLRootElement(file);
             calendarElements.calendarName = getText(getRequiredChild(root, "name"));
             calendarElements.timeZone = TimeZoneAliases.zoneId(getText(getRequiredChild(root, "timeZone")));
-            calendarElements.description = getText(getRequiredChild(root, "description"));
-            calendarElements.firstValidDate =
-                    DateTimeUtils.parseLocalDate(getText(getRequiredChild(root, "firstValidDate")));
-            calendarElements.lastValidDate =
-                    DateTimeUtils.parseLocalDate(getText(getRequiredChild(root, "lastValidDate")));
+            calendarElements.description = getText(root.getChild("description"));
             calendarElements.holidays = parseHolidays(root, calendarElements.timeZone);
+            final String firstValidDateStr = getText(root.getChild("firstValidDate"));
+            calendarElements.firstValidDate = DateTimeUtils.parseLocalDate(
```

Comment:
Shouldn't need to parse in the null case; can just use the min value, yeah? Same w/ max.

Reply:
I think you are asking if `firstValidDate` even needs to be present.  It does.  The calendar could be valid before the first holiday.

Reply:
I'm saying when the element is null, we shouldn't need to take go from LocalDate -> String -> LocalDate, ie:

`DateTimeUtils.parseLocalDate(Collections.min(calendarElements.holidays.keySet()).toString())`

Reply:
`Collections.min(calendarElements.holidays.keySet())` should suffice.